modules/mono/csharp_script.cpp

```diff
@@ -3117,13 +3119,10 @@ MethodInfo CSharpScript::get_method_info(const StringName &p_method) const {
 }
 
 Error CSharpScript::reload(bool p_keep_state) {
-	bool has_instances;
-	{
-		MutexLock lock(CSharpLanguage::get_singleton()->script_instances_mutex);
-		has_instances = instances.size();
-	}
-
-	ERR_FAIL_COND_V(!p_keep_state && has_instances, ERR_ALREADY_IN_USE);
+	
+	if(!reload_invalidated) return OK;
+	
+	reload_invalidated = false; /* In the case of C#, reload doesn't really do any script reloading. That's done separately via domain reloading. */
```

Comment:
We prefer `//` comments to `/* */` comments:

```suggestion
	// In the case of C#, reload doesn't really do any script reloading.
	// That's done separately via domain reloading.
	reload_invalidated = false;
```